src/com/boxboat/jenkins/library/buildVersions/GitBuildVersions.groovy

```diff
@@ -28,40 +28,92 @@ class GitBuildVersions implements Serializable {
         gitRepo = Config.gitAccount.checkoutRepository(Config.global.git.buildVersionsUrl, targetDir, 1)
     }
 
-    def setEventImageVersion(String event, Image image, String version) {
+    def setEventImageVersion(String event, Image image, String version, Map<Object, Object> metadata = null) {
         def dir = "${gitRepo.dir}/image-versions/${event}"
-        def path = "${dir}/${Utils.alphaNumericDashLower(image.path)}.txt"
+        def path = "${dir}/${Utils.alphaNumericDashLower(image.path)}"
+        def txtFile = "${path}.txt"
+        def yamlFile = "${path}.yaml"
+        // conversion from text to yaml
+        if (Config.pipeline.fileExists(txtFile)) {
+            Config.pipeline.sh """
+                rm "${txtFile}"
+            """
+        }
+        Map<String, Object> data = new LinkedHashMap<>()
+        data.put("version", version)
+        if(metadata != null && !metadata.isEmpty()) {
+            data.put("metadata", metadata)
+        }
+        def yamlStr = YamlUtils.dump(data)
```

Comment:
The reason not to use the `pipeline-utility-plugin` would be to keep `jenkins-dockhand` plugin dependencies to a minimum.... However, the `pipeline-utility-steps` is already a requirement because `readJSON` is being used elsewhere within the library. `readYaml` and `writeYaml` uses SnakeYAML under the hood, which is what `YamlUtils` uses so I'll test to ensure that the `pipeline-utility-steps` function the same way

Reply:
`writeYaml` doesn't function quite the way we want it to - so we'll use `readYaml` and continue to use SnakeYAML directly for writing the yaml files